In [1]:
import pandas as pd
from business_tools import (
    calculate_commission,
    calculate_total_premium,
    filter_policies_by_state,
    load_insurance_sales,
)


In [2]:
!python --version

Python 3.10.0


## Step 1: Create the Model Context Protocol Server

In [3]:
from fastmcp import FastMCP

# Initialize MCP Server
mcp = FastMCP("BusinessTools")

@mcp.tool()
def calculate_profit(revenue: float, expenses: float) -> float:
    """Calculate profit."""
    return revenue - expenses

print("MCP Server ready! Tools: calculate_profit.")

MCP Server ready! Tools: calculate_profit.


## Step 2: Test the MCP Tool Locally

In [4]:
profit = calculate_profit(1000, 600)
print("Profit:", profit)

Profit: 400


## Step 3: Expand MCP by Adding More Functions

In [5]:

@mcp.tool()
def get_sales_from_csv(file_path: str) -> float:
    """Read total sales from a CSV file."""
    df = pd.read_csv(file_path)
    return df["sales"].sum()

## Step 4: Create the CSV File

In [7]:
csv_data = """date,sales
2024-02-26,200
2024-02-27,450
2024-02-28,300
"""

# Save to a file in the data folder
with open("data/sales_data.csv", "w") as file:
    file.write(csv_data)

print("CSV file created successfully!")

CSV file created successfully!


## Step 5: Run get_sales_from_csv

In [8]:
sales_total = get_sales_from_csv("data/sales_data.csv")
print("Total Sales:", sales_total)

Total Sales: 950


## Step 6: Calculate Commission

In [ ]:
commission = calculate_commission([300, 700, 200], rate=0.1)
print('Commission:', commission)

## Step 7: Load `insurance_sales.csv`

In [ ]:
records = load_insurance_sales('data/insurance_sales.csv')
print(f'Total records: {len(records)}')

In [ ]:
total_premium = calculate_total_premium(records)
print('Total Premium:', total_premium)

In [ ]:
ca_policies = filter_policies_by_state(records, 'CA')
print('CA Policies:', len(ca_policies))

These steps show how quickly you can tally premiums and isolate policies by state—handy for reviewing territory performance and carrier payouts.

# 🚀 Super Simple Summary: MCP Project Overview

## **What We Did** 🛠️
1. **Set up our Python environment**  
   - Installed Python 3.10  
   - Created a virtual environment (venv)  
   - Installed necessary packages (like pandas)  
   - Ensured Jupyter Notebook was using the correct environment  

2. **Created a simple MCP (Model Context Protocol) server**  
   - Used `FastMCP` to set up a tool that can process functions  
   - Created a `calculate_profit` function to subtract expenses from revenue  
   - Registered that function with MCP  

3. **Expanded the MCP tool with more functionality**  
   - Added a new function `get_sales_from_csv` to read a CSV file and sum the sales column  
   - Used `pandas` to load and process the CSV data  

4. **Tested everything** ✅  
   - Created a CSV file inside the notebook  
   - Ran our function to sum the sales  
   - Verified that the result was correct (`950` total sales)  

---

## **Why Is This Important?** 🔥
- **MCP helps structure tools for AI models** 🧠  
  → Instead of just writing random functions, MCP lets us build structured "tools" that can be reused.  

- **We practiced file handling, data processing, and function registration** 📂  
  → This is a **real-world skill** that applies to **machine learning, AI automation, and software development**.  

---

## **Next Steps:**
Would you like to:  
1️⃣ **Expand this project** (e.g., more data analysis, visualization)?  
2️⃣ **Start a new MCP-based project**?  
3️⃣ **Dive deeper into MCP internals** (how it works under the hood)?  

🚀🔥 Let’s build something amazing!